indexer/services/roundtable/src/tasks/aggregate-trading-rewards.ts

```diff
@@ -213,23 +229,305 @@ export class AggregateTradingReward {
+      logger.info({
+        at: 'aggregate-trading-rewards#updateTradingRewardsAggregation',
+        message: 'Updated trading rewards aggregation',
+        start: interval.start.toISO(),
+        end: interval.end.toISO(),
+      });
+    } catch (error) {
+      await Transaction.rollback(txId);
+      logger.info({
+        at: 'aggregate-trading-rewards#updateTradingRewardsAggregation',
+        message: 'Failed to update trading rewards aggregation',
+        error: error.message,
+        start: interval.start.toISO(),
+        end: interval.end.toISO(),
+      });
+      throw error;
+    }
+  }
+
+  /**
+   * Generate all update and create objects for the interval, by aggregating all trading rewards
+   * for the interval.
+   */
+  private async getAggregationUpdateAndCreateObjectsFromInterval(
+    interval: Interval,
+    intervalTradingRewardsByAddress: IntervalTradingRewardsByAddress,
+  ): Promise<AggregationUpdateAndCreateObjects> {
+    const tradingRewardAddresses: string[] = Object.keys(intervalTradingRewardsByAddress);
+
+    const existingAggregateTradingRewards: TradingRewardAggregationFromDatabase[] = await
+    TradingRewardAggregationTable.findAll({
+      addresses: tradingRewardAddresses,
+      period: this.period,
+    }, []);
+    const existingAggregateTradingRewardsMap:
+    { [address: string]: TradingRewardAggregationFromDatabase } = _.keyBy(
+      existingAggregateTradingRewards,
+      TradingRewardAggregationColumns.address,
+    );
+
+    const startedAt: string = this.getStartedAt(interval);
+    const startedAtHeight: string = await this.getNextBlock(startedAt);
+    const aggregateTradingRewardsToUpdate: TradingRewardAggregationUpdateObject[] = _.intersection(
+      tradingRewardAddresses,
+      Object.keys(existingAggregateTradingRewardsMap),
+    ).map((address: string) => {
+      return {
+        id: TradingRewardAggregationTable.uuid(address, this.period, startedAtHeight),
+        amount: Big(intervalTradingRewardsByAddress[address])
+          .plus(existingAggregateTradingRewardsMap[address].amount).toFixed(),
+      };
+    });
+
+    const aggregateTradingRewardsToCreate: TradingRewardAggregationCreateObject[] = _.difference(
+      tradingRewardAddresses,
+      Object.keys(existingAggregateTradingRewardsMap),
+    ).map((address: string) => {
+      return {
+        address,
+        startedAt,
+        startedAtHeight,
+        period: this.period,
+        amount: intervalTradingRewardsByAddress[address],
+      };
+    });
+
+    return {
+      updateObjects: aggregateTradingRewardsToUpdate,
+      createObjects: aggregateTradingRewardsToCreate,
+    };
+  }
+
+  private getStartedAt(interval: Interval): IsoString {
+    return interval.start.startOf(this.getDateTimeUnit()).toISO();
+  }
+
+  private async getNextBlock(time: IsoString): Promise<string> {
+    const blocks: BlockFromDatabase[] = await BlockTable.findAll({
+      createdOnOrAfter: time,
+      limit: 3,
```

Comment:
why is the limit 3 here?

Reply:
oops that was while I was testing, I changed it to 1